zap/src/main/java/org/parosproxy/paros/control/Control.java

```diff
@@ -666,4 +668,43 @@ public void setMode(Mode mode) {
             model.getOptionsParam().getViewParam().setMode(mode.name());
         }
     }
+
+    public static LinkedList<String> getRecentSessionList() {
+        LinkedList<String> recentSessionList = new LinkedList<>();
+        for (String path :
+                Model.getSingleton()
+                        .getOptionsParam()
+                        .getConfig()
+                        .getStringArray("recentsessions.path")) {
+            recentSessionList.add(path);
+        }
+        return recentSessionList;
+    }
+
+    private void setRecentSessionList(String path) {
+        LinkedList<String> recentSessionList = getRecentSessionList();
+        int index = recentSessionList.indexOf(path);
+        if (index == 0) {
+            return;
+        } else if (index > 0) {
+            recentSessionList.remove(index);
+        }
+        recentSessionList.addFirst(path);
+        while (recentSessionList.size() > 10) {
+            recentSessionList.removeLast();
+        }
+        try {
+            FileConfiguration config = model.getOptionsParam().getConfig();
+            config.clearProperty("recentsessions.path");
+            for (int i = 0; i < recentSessionList.size(); ++i) {
+                config.setProperty("recentsessions.path(" + i + ")", recentSessionList.get(i));
+            }
```

Comment:
This could use an advanced for or probably even forEach.
```java
            for (String sessionPath : getRecentSessionList()) {
                config.setProperty("recentsessions.path(" + sessionPath + ")", sessionPath);
            }
```

Reply:
This doesn't work for me with an error:
Caused by: org.w3c.dom.DOMException: INVALID_CHARACTER_ERR: An invalid or illegal XML character is specified.
I think it's best to keep it simple as in the original Paros code and the apache commons configuration doc.

Reply:
I think the path contains INVALID CHARACTERS for a key.

Reply:
we shouldnt use arbitrary characters for the keys, and we should check the values are suitably encoded ...

Reply:
I didn't get it :( do you mean I need to encode the path?